experimental/Schemes/CoveredScheme.jl

```diff
@@ -568,3 +568,162 @@ function CoveredClosedEmbedding(X::AbsCoveredScheme, I::IdealSheaf;
+  if !isdefined(f, :composed_map)
+    result = underlying_morphism(first(maps(f)))::CoveredSchemeMorphism
+    for i in 2:length(maps(f))
+      result = compose(result, underlying_morphism(maps(f)[i]))::CoveredSchemeMorphism
+    end
+    f.composed_map = result
+  end
+  return f.composed_map::CoveredSchemeMorphism
+end
+
+### Specialized functionality
+
+# Casting into the minimal concrete type for AbsCoveredSchemeMorphism
+function CoveredSchemeMorphism(f::CompositeCoveredSchemeMorphism)
+  return underlying_morphism(f)
+end
+
+function CoveredSchemeMorphism(f::CoveredSchemeMorphism)
+  return f
+end
+
+########################################################################
+# The standard constructors
+########################################################################
+@doc raw"""
+    composite_map(f::AbsCoveredSchemeMorphism, g::AbsCoveredSchemeMorphism)
+
+Realize the composition ``x → g(f(x))`` as a composite map, i.e. an 
+instance of `CompositeCoveredSchemeMorphism`. 
+
+# Examples
+```jldoctest
+julia> IA2 = affine_space(QQ, [:x, :y])
+Affine space of dimension 2
+  over rational field
+with coordinates [x, y]
+
+julia> (x, y) = gens(OO(IA2));
+
+julia> I = ideal(OO(IA2), [x, y]);
+
+julia> pr = blow_up(IA2, I);
+
+julia> JJ = ideal_sheaf(exceptional_divisor(pr));
+
+julia> inc_E = oscar.CoveredClosedEmbedding(domain(pr), JJ);
+
+julia> comp = oscar.composite_map(inc_E, pr)
+Composite morphism of
+  Morphism: scheme over QQ covered with 2 patches -> scheme over QQ covered with 2 patches
+  Blow-up: scheme over QQ covered with 2 patches -> scheme over QQ covered with 1 patch
+
+julia> oscar.maps(comp)[1] === inc_E
+true
+
+julia> oscar.maps(comp)[2] === pr
+true
+
+```
+"""
+function composite_map(f::AbsCoveredSchemeMorphism, g::AbsCoveredSchemeMorphism)
+  return CompositeCoveredSchemeMorphism([f, g])
+end
+
+function composite_map(f::AbsCoveredSchemeMorphism, g::CompositeCoveredSchemeMorphism)
+  return CompositeCoveredSchemeMorphism(pushfirst!(Vector{AbsCoveredSchemeMorphism}(copy(maps(g))), f))
+end
+
+function composite_map(f::CompositeCoveredSchemeMorphism, g::CompositeCoveredSchemeMorphism)
+  return CompositeCoveredSchemeMorphism(vcat(maps(f), maps(g)))
+end
+
+function composite_map(f::CompositeCoveredSchemeMorphism, g::AbsCoveredSchemeMorphism)
+  return CompositeCoveredSchemeMorphism(push!(Vector{AbsCoveredSchemeMorphism}(copy(maps(f))), g))
+end
+
+########################################################################
+# Printing
+########################################################################
+function Base.show(io::IO, f::CompositeCoveredSchemeMorphism)
+  io = pretty(io)
+  if get(io, :supercompact, false)
+    print(io, "Composite morphism")
+  else
+    map_string = "$(domain(f)) -> "
+    for i in 2:length(maps(f))
+      map_string = map_string * "$(domain(map(f)[i])) -> "
+    end
+    map_string = map_string * "$(codomain(map(f)[end]))"
+
+    print(io, "Composition of ", map_string)
```

Comment:
No need to print to an intermediate string
```suggestion
    print(io, "Composition of ", map_string)
    print(io, domain(f), "  -> ")
    for i in 2:length(maps(f))
      print(io, domain(map(f)[i]), " -> ")
    end
    print(io, codomain(map(f)[end]))
```